src/models/v2/server.ts

```diff
@@ -0,0 +1,43 @@
+import { Mixin } from '../utils';
+import { BaseModel } from '../base';
+import { ServerInterface } from '../server';
+import { DescriptionMixin } from './mixins/description';
+import { BindingsMixin } from './mixins/bindings';
+
+export class Server extends Mixin(BaseModel, DescriptionMixin, BindingsMixin) implements ServerInterface {
+    id(): string {
+        return this.json('id');
+    }
```

Comment:
We had discussion in Slack that `id` should not be a part of `_json` :) I also gave you example how it should work - https://github.com/asyncapi/parser-js/blob/next-major/src/models/v2/mixins/extensions.ts#L11

```suggestion
  constructor(
    private readonly _id: string,
    _json: Record<string, any>,
  ) {
    super(_json);
  }

  id(): string {
    return this._id;
  }
```

Reply:
I did this because I modeled the utils function to convert object to array something like this 

```
export function createArrayFromMap(json: Record<string,any>){
  const ArrayObject = [];
  for (const [key, value] of Object.entries(json)) {
    value['id'] = key;
    ArrayObject.push(value);
  };

  return ArrayObject;
}
```

so I thought maybe I don't need the add a separate id variable can just get it to format it. Main reason was so that the `v2` and `v3` have minimal changes, because to add seperate `_id` parameter now we have to pass something like `new Servers(id, this._json)`

Reply:
v3 changes may be much bigger than they are now (to be honest we don't have anything approved so v3=v2 now), so please don't look at this proposal from Fran as something final.

About `id`. Have in mind that if someone will run `json()` should get the value that is defined for the object in the specification (not including traits because we have to combine them) without any additional fields that are not defined in the object, so what you want to get with adding the `id` field is nothing more than changing the original json. I know that adding an id as an argument to a constructor can be weird, but in this case the id is outside the object, and in v3 it can be part of the object and it will be easier to handle.